semantic_model_generator/snowflake_utils/env_vars.py

```diff
@@ -35,15 +38,16 @@ def assert_required_env_vars() -> list[str]:
         missing_env_vars.append("SNOWFLAKE_ACCOUNT_LOCATOR")
     if not SNOWFLAKE_HOST:
         missing_env_vars.append("SNOWFLAKE_HOST")
-    if not SNOWFLAKE_PASSWORD and not SNOWFLAKE_AUTHENTICATOR:
-        missing_env_vars.append("SNOWFLAKE_PASSWORD/SNOWFLAKE_AUTHENTICATOR")
+    if not SNOWFLAKE_AUTHENTICATOR:
+        if not SNOWFLAKE_PASSWORD and not SNOWFLAKE_PRIVATE_KEY_FILE:
```

Comment:
I'm assuming `SNOWFLAKE_PASSWORD` and `SNOWFLAKE_PRIVATE_KEY_FILE` are mutually exclusive (i.e. you can have one or the other but not both)? If that's the case, it might be nice to make the message a little clearer to the user:

```suggestion
        if not SNOWFLAKE_PASSWORD:
                  missing_env_vars.append("SNOWFLAKE_PASSWORD/SNOWFLAKE_AUTHENTICATOR")
        elif not SNOWFLAKE_PRIVATE_KEY_FILE:
                  missing_env_vars.append("SNOWFLAKE_PRIVATE_KEY_FILE")
```